pkg/executables/flux.go

```diff
@@ -63,17 +58,62 @@ func (f *Flux) BootstrapToolkitsComponentsGithub(ctx context.Context, cluster *t
 
 	_, err = f.ExecuteWithEnv(ctx, env, params...)
 	if err != nil {
-		return fmt.Errorf("executing flux bootstrap: %v", err)
+		return fmt.Errorf("executing flux bootstrap github: %v", err)
 	}
 
 	return err
 }
 
-// BootstrapToolkitsComponentsGit creates the Git repository if it does not exist, and commits the toolkit
-// components manifests to the main branch. Then it configures the target cluster to synchronize with the repository.
-// If the toolkit components are present on the cluster, the bootstrap command will perform an upgrade if needed.
+// BootstrapToolkitsComponentsGit commits the toolkit components manifests to the branch of a Git repository.
+// It then configures the target cluster to synchronize with the repository. If the toolkit components are present on the cluster, the
+// bootstrap command will perform an upgrade if needed.
 func (f *Flux) BootstrapToolkitsComponentsGit(ctx context.Context, cluster *types.Cluster, fluxConfig *v1alpha1.FluxConfig) error {
-	return nil
+	c := fluxConfig.Spec
+	params := []string{
+		"bootstrap",
+		gitProvider,
+		"--url", c.Git.RepositoryUrl,
+		"--username", c.Git.Username,
+		"--path", c.ClusterConfigPath,
+		"--ssh-key-algorithm", privateKeyAlgorithm,
+	}
```

Comment:
Why not make these part of the common private method as well?
Maybe we could simplify all this code by having some kind of `bootstrap` command builder construct.

like
```go
newBoostrapCommand(gitProviderm,
  withUrl(c.Git.RepositoryUrl),
  withUsername(c.Git.Username),
  ...
  withPassword(password),
)
```

Reply:
The parameters for both bootstrap commands are a little different, so I pulled out the common params in a different method.

Reply:
To be fixed: https://github.com/aws/eks-anywhere/issues/1952